openedx/core/lib/xblock_utils.py

```diff
@@ -302,7 +302,9 @@ def add_staff_markup(user, has_instructor_access, disable_staff_debug_info, bloc
         # check that the course is a mongo backed Studio course before doing work
         is_studio_course = block.course_edit_method == "Studio"
 
-        if is_studio_course:
+        is_ccx_course = hasattr(course_id, 'ccx')
+        # preventing users from opening ccx courses in Studio, since Studio doesn't support CCX courses
+        if is_studio_course and not is_ccx_course:
             # build edit link to unit in CMS. Can't use reverse here as lms doesn't load cms's urls.py
             edit_link = "//" + settings.CMS_BASE + '/container/' + unicode(block.location)
```

Comment:
@felipevolpone: I believe you can just change `lms/djangoapps/ccx/overrides.py:get_override_for_ccx` so that it returns `"CCX"` when the `name` passed in is `course_edit_method`. One of the main architectural goals of CCX is to keep CCX-specific knowledge from leaking out into other systems as much as possible, and to maintain the facade of "it's just another course" to querying systems. Since there is already a field toggle for, "Can we edit this in Studio?", I'd rather CCX make use of that, rather than having more CCX-specific handling code make it out to this layer.

As to whether it's required for merging -- that mostly depends on whether what I just described works, or if there are hidden complexities that I'm unaware of. I strongly prefer putting this in the CCX app instead of adding anything to `module_render()` or `add_staff_markup()`, but please let me know if this is not as straightforward as I think.

Thank you.